conda/models/index_record.py

```diff
@@ -54,12 +54,31 @@ def box(self, instance, val):
 
 class TimestampField(NumberField):
 
-    def box(self, instance, val):
-        val = super(TimestampField, self).box(instance, val)
-        if val and val > 253402300799:  # 9999-12-31
-            val /= 1000  # convert milliseconds to seconds; see conda/conda-build#1988
+    # @staticmethod
+    # def _make_seconds(val):
+    #     if val:
+    #         val = int(val)
+    #         if val > 253402300799:  # 9999-12-31
+    #             val //= 1000  # convert milliseconds to seconds; see conda/conda-build#1988
+    #     return val
+
+    @staticmethod
+    def _make_milliseconds(val):
+        if val:
+            val = int(val)
```

Comment:
This is truncating any floating point numbers.  You should probably multiply prior to calling int.